### PR TITLE
Alow setting and sending of a resource pack to players

### PR DESCRIPTION
**Pull request by Aaron1011**
This relies on GlowstoneMC/Glowkit#49

---

_Edited by turt2live_

**Related Links:**
- Glowkit: GlowstoneMC/Glowkit#49
- No ticket (none needed)

### CODE REVIEW

File: src/main/java/net/glowstone/GlowServer.java [hunk 1]
```diff
@@ -750,6 +750,12 @@ public boolean getProxySupport() {
         return config.getBoolean(ServerConfig.Key.PROXY_SUPPORT);
     }
 
+    /**
+     * Get the resource pack url for this server, or <code>null</code> if not set
+     * @return url The url of the resource pack to use, or <code>null</code>
+     */
+    public String getResourcePackURL() { return config.getString(ServerConfig.Key.RESOURCE_PACK); }
```
Comment (kitskub) [inline src/main/java/net/glowstone/GlowServer.java:792]: Look at the other methods in this class and fix formatting.

Comment (turt2live) [inline src/main/java/net/glowstone/GlowServer.java:792]: Formatting is not important at this stage

**Comment by Aaron1011** [comment]
`GLOWKIT: 49`

**Comment by dequis** [comment]
@Aaron1011 please also edit the description to mention GlowstoneMC/Glowkit#49, to fix travis builds 

(hey, it's not my fault turt2live decided to change the way turt2bot picks up linked PRs while my travis branch was awaiting review)

**Comment by bendem** [comment]
#blamedequis

**Comment by dequis** [comment]
Simple conflict in GlowServer, keep both sides (and fix javadoc). [Branch](https://github.com/dequis/Glowstone/compare/pr-500)

**Comment by turt2live** [comment]
@Aaron1011 Just curious, how does the vanilla server handle the resource pack hash? Can this be set in the `server.properties` or is it magically determined? This looks pretty good for testing as-is though.

**Comment by Aaron1011** [comment]
@turt2live: It's just set in `server.properties`. Vanilla makes no attempt to compute it if it's not present.

**Comment by turt2live** [comment]
Interesting. Thanks for the info!

**Comment by turt2live** [comment]
For reference when testing: This PR is not broken, the client is. The client reports success on almost all cases instead of failure when it should. It also doesn't validate very much, so it is possible to get the client to display a never ending "Making request" screen (press escape to exit).

File: src/main/java/net/glowstone/GlowServer.java [hunk 2]
```diff
@@ -785,6 +785,18 @@ public boolean useRconColors() {
         return config.getBoolean(ServerConfig.Key.RCON_COLORS);
     }
 
+    /*
+     * Get the resource pack url for this server, or <code>null</code> if not set
+     * @return url The url of the resource pack to use, or <code>null</code>
+     */
+    public String getResourcePackURL() { return config.getString(ServerConfig.Key.RESOURCE_PACK); }
```
Comment (turt2live) [inline src/main/java/net/glowstone/GlowServer.java:792]: These one line methods counter the rest of the file. Just a note for whoever pulls this as it's not urgent

**Comment by SpaceManiac** [comment]
Merged in 66295d01292318c2b4bf1603f170a1304de2298f.

**Pull request closed without merging**